src/test/ui/associated-consts/assoc-const-ty-mismatch.stderr

```diff
@@ -8,7 +8,7 @@ note: associated constant defined here does not match type
   --> $DIR/assoc-const-ty-mismatch.rs:5:3
    |
 LL |   const N: usize;
-   |   ^^^^^^^^^^^^^^^
+   |   ^^^^^^^
```

Comment:
Should this stop at the ident, or go until the end of the type?

Reply:
I'd personally include the type (but not initializer).
Also bounds for traits.

All of those are parts of an item "signature" (as opposed to item body).
The only thing I'm not sure about is whether to include `where` clauses to this span.